### ISSUE TITLE
Reverse Distillation — Custom training, validation, and legacy testing

### ISSUE THREAD

**Issue by Tiff223** [Tiff223/Leather_inspection#19]
## Purpose

Fit Reverse Distillation on the confirmed-good custom leather training set, calibrate it with the independent custom validation collection, and evaluate it on the legacy common custom test.

This issue is one track of the model's three-track leather experiment. Benchmark, custom-trained, and MVTec-to-custom transfer results must remain separate and must never be averaged.

## Model-specific recording requirements

Record pretrained encoder/backbone, feature layers, anomaly-map combination mode, input size, epochs, optimizer settings, device, and all threshold/post-processing values.

## Pipeline preparation

Before execution:

- [ ] Create the MVTec and custom configuration files in the appropriate `configs/mvtec/` and `configs/custom_leather/` folders
- [ ] Implement and verify every required evaluator and report adapter
- [ ] Register `reverse_distillation` in `pipeline/leather_pipeline.json`
- [ ] Confirm every stage is `READY`; do not bypass a `NOT WIRED` stage
- [ ] Run dataset preflight and confirm no cross-split byte duplicates
- [ ] Preview the full master pipeline

Master pipeline preview:

```bash
python pythons/leather_three_track_pipeline.py plan --model reverse_distillation \
  --track all --run-id reverse_distillation_three_track_run01
```

Master pipeline execution:

```bash
python pythons/leather_three_track_pipeline.py run --model reverse_distillation \
  --track all --run-id reverse_distillation_three_track_run01 --execute
```

If execution stops, fix the cause and repeat the exact command with the same master run ID. Completed stages must be skipped and preserved.

## Dataset protocol

- Training: 199 confirmed-good custom leather images
- Independent validation: 31 images — 20 good, 6 cut, 5 stained
- Legacy test: 33 images — 23 good, 5 cut, 5 stained
- Validation and test images must not enter representation fitting
- Freeze the model and validation-selected operating point before evaluating the test
- The legacy test has been examined previously and is not an untouched final test

## Track checklist

- [ ] Preview and execute `custom_fit_validate`
- [ ] Verify checkpoint, master/child ledgers, configuration snapshot, and SHA-256 dataset manifest
- [ ] Confirm exactly which parameters are fitted and which weights are pretrained/frozen
- [ ] Record validation AUROC, AP, threshold, confusion matrix, mistakes, and score ranges
- [ ] Preview and execute `custom_evaluate` using the frozen checkpoint and validation decision rule
- [ ] Record test AUROC, AP, classification metrics, mistakes, score ranges, and runtime
- [ ] Generate heat maps/overlays for mistakes
- [ ] Preview and execute `custom_report`
- [ ] Generate and integrity-check the DOCX report
- [ ] Execute the custom `update_comparison`
- [ ] Update the custom overall table, validation table, score-range table, chart, registry, and dashboard
- [ ] Verify no benchmark or transfer result was merged into the custom ranking

## Artifacts and results

- Master run ID: `reverse_distillation_three_track_run01`
- Configurations:
- Checkpoint:
- Validation threshold and provenance:
- Dataset manifest:
- Stage ledgers:
- Validation/test summaries:
- Per-image scores and mistake figures:
- Report:
- Comparison files:

## Limitations

Pixel metrics may be reported only if matching custom ground-truth masks exist. Describe the 33-image result as a legacy common-test comparison, not an unbiased final deployment estimate.

## Conclusion

Not completed yet. Close only after the local artifacts and comparison placement have been reviewed.
